ArrayList/ArrayList.cs

```diff
@@ -1,8 +1,580 @@
 ﻿using System;
+using System.Text;
 
-namespace ArrayList
+namespace List
 {
-    public class ArrayList
+    public class ArrayList<T> where T : IComparable<T>
     {
+        public int Length
+        {
+            get
+            {
+                return _length;
+            }
+            private set
+            {
+                if (value >= 0)
+                {
+                    _length = value;
+                }
+                else if (value < 0)
+                {
+                    _length = 0;
+                }
+            }
+        }
+
+        public T this[int index]
+        {
+            get
+            {
+                if (index >= 0 && index < Length)
+                {
+                    return _array[index];
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+            set
+            {
+                if (index >= 0 && index < _array.Length)
+                {
+                    _array[index] = value;
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+        }
+
+        private int _length;
+        private T[] _array;
+        private int _initLength = 10;
+
+        public ArrayList()
+        {
+            Length = 0;
+            _array = new T[_initLength];
+        }
+
+        public ArrayList(T value)
+        {
+            if (value != null)
+            {
+                Length = 1;
+                _array = new T[_initLength];
+                _array[0] = value;
+            }
+            else
+            {
+                throw new ArgumentException("Values is null");
+            }
+        }
+
+        public ArrayList(T[] values)
+        {
+            if (values != null)
+            {
+                Length = values.Length;
+                _array = new T[(int)(values.Length * 2)];
+                for (int i = 0; i < Length; i++)
+                {
+                    _array[i] = values[i];
+                }
+            }
+            else
+            {
+                throw new ArgumentException("Array is null");
+            }
+        }
+
+        /// <summary>
+        /// Adds a value to the end of the list.
+        /// </summary>
+        /// <param value="value">Value to be added.</param>
+        public void Add(T value)
+        {
+            int lastIndex = Length;
+            AddByIndex(value, lastIndex);
```

Comment:
Try to remove `lastIndex` variable. You can find solution in inteliSense to keep it readable without creating variable.

Reply:
Ok